src/poprox_storage/repositories/experiments.py

```diff
@@ -182,10 +182,21 @@ def fetch_active_expt_recommender_urls(self, date: datetime.date | None = None)
         return recommender_lookup_by_group
 
     def fetch_active_expt_assignments(self, date: datetime.date | None = None) -> dict[UUID, Assignment]:
-        assignments_tbl = self.tables["expt_assignments"]
-
         group_ids = self.fetch_active_expt_group_ids(date)
+        group_lookup_by_account = self._fetch_assignments_by_group_ids(group_ids)
+
+        return group_lookup_by_account
+
+    def fetch_assignments_by_experiment_id(self, experiment_id: UUID) -> dict[UUID, Assignment]:
+        experiment = self.fetch_experiment_by_id(experiment_id)
 
+        group_ids = [group.id for group in experiment.groups]
```

Comment:
I don't think fetch_experiment_by_id returns groups.

Reply:
(sorry for taking so long to notice this issue. I was doubling back to review PRs before the Tuesday meeting.)
I might want @karlhigley's double-check here, but the implementation of `fetch_experiment_by_id` that  currently see sets the phases to `[]` an I think `.groups` pulls the group info out of the phase dictionary right? I'm not missing somewhere where this info is filled in later right?

Reply:
I don't think we've had any reason to fetch an experiment with the whole object graph pulled along with it before, so yeah, this won't currently ever return the groups. We could modify the experiment fetching method to pull in the phases etc though

Reply:
I think that would be my current proposal for how to make this work. I know that' a few extra queries, but in the cosmic scale, I think doing that work every time is better than having lots of small functions for what parts of the object graph we want and having to remember if an experiment view model is complete or not.